Source/LuaBridge/RefCountedPtr.h

```diff
@@ -159,13 +159,17 @@ class RefCountedPtr : private detail::RefCountedPtrBase
     */
     void reset()
     {
-        if (m_p != 0)
+        auto myCount_it = getRefCounts().find(m_p); // use find() to avoid adding a spurious pointer to getRefCounts()
```

Comment:
Could you please clarify what does the spurious pointer mean?
Everything we do here is theoretically operations on variables with the `auto` storage.

Reply:
We do not want to insert *any* value (including null) into the `unordered_map` when what we're doing is trying to remove a value if it's there. The `find` function finds it without inserting.

Reply:
EDIT: I realize my previous comment makes no sense. I changed the comment to reflect the fact that using `find` allows us to only look up `m_p` once. We need to find out a) if it exists and b) whether the count has dropped to zero, which is potentially two lookups if we didn't use `find`.

The basic issue is that it is completely possible (and completely okay as far as Lua is concerned) to have code like this:

```c++
return RefCountedPtr<MyType> ( nullptr );
```
My code actually has a function declaration inside a container class that looks more like this

```c++
RefCountedPtr<MyType> CreateMyTypeIfItExists(); // return nullptr if MyType doesn't exist.
```
In Lua it looks like

```lua
local var = container:CreateMyTypeIfItExists()
if var then
   -- do something
end
```